cmd/oras/internal/option/target.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/oci"
+	"oras.land/oras-go/v2/registry"
+	"oras.land/oras/cmd/oras/internal/fileref"
+)
+
+const (
+	TargetTypeRemote    = "registry"
+	TargetTypeOCILayout = "oci-layout"
+)
+
+// Unary target option struct.
+type Target struct {
+	Remote
+	RawReference string
+	Type         string
+	Reference    string //contains tag or digest
+
+	isOCI bool
+}
+
+// ApplyFlags applies flags to a command flag set for unary target
+func (opts *Target) ApplyFlags(fs *pflag.FlagSet) {
+	opts.applyFlagsWithPrefix(fs, "", "")
+	opts.Remote.ApplyFlags(fs)
+}
+
+// AnnotatedReference returns full printable reference.
+func (opts *Target) AnnotatedReference() string {
+	return fmt.Sprintf("[%s] %s", opts.Type, opts.RawReference)
+}
+
+// applyFlagsWithPrefix applies flags to a command flag set with a prefix string.
+// The full target flag should be provided as:
+//
+//	--target type={oci-layout|registry}.
+//
+// Since only OCI image layout is supported, a short boolean flag
+// `--[{from|to}-oci-layout]` is used to simplify UX.
+//   - If the flag is set, it equals to `--target type=oci-layout`;
+//   - If not, it equals to `--target type=registry`
```

Comment:
Consider the following documentation:
```
The complete form of the `target` flag is designed to be

    --target type=<type>[[,<key>=<value>][...]]

For better UX, the boolean flag `--oci-layout` is introduced as an alias of `--target type=oci-layout`.
Since there is only one target type besides the default `registry` type, the full form is not implemented until a new type comes.
```